experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -1,184 +1,187 @@
-    this.inputs = findInputs(queryExecution.logical());
-    this.outputs = findOutputs(queryExecution.logical());
-    print("%*** -> logical: ", queryExecution.logical());
-    print("%*** -> physical: ", queryExecution.executedPlan());
-    print("sparkPlanInfo Exec:"+ startEvent.executionId()+ "  ", sparkPlanInfo);
-    this.runId = marquezContext.startRun(startEvent.time(), inputs, outputs, extraInfo.toString());
   }
 
-  private List<String> findInputs(LogicalPlan logical) {
-    List<String> result = new ArrayList<String>();
-    Collection<LogicalPlan> leaves = JavaConversions.asJavaCollection(logical.collectLeaves());
-    for (LogicalPlan leaf : leaves) {
-        if (leaf instanceof LogicalRelation) {
-          LogicalRelation lr = (LogicalRelation) leaf;
-          BaseRelation lrr = lr.relation();
-          if (lrr instanceof HadoopFsRelation) {
-            FileIndex location = ((HadoopFsRelation)lrr).location();
-            Collection<Path> rootPaths = JavaConversions.asJavaCollection(location.rootPaths());
-            for (Path rootPath : rootPaths) {
-              result.add(rootPath.toUri().toString());
-            }
-          } else {
-            extraInfo.append("unknown baseRelation in leaf: ").append(leaf.getClass().getName())
-            .append("\n  ").append(leaf)
-            .append("\n  ").append(lrr.getClass().getName()).append(" ").append(lrr).append("\n");
-          }
-        }else {
-          extraInfo.append("unknown leaf node: ").append(leaf.getClass().getName())
-          .append("\n  ").append(leaf).append("\n");
-        }
-    }
-    return result;
+  public void end(SparkListenerSQLExecutionEnd endEvent) {
   }
 
-  private List<String> findOutputs(LogicalPlan logical) {
-    if (logical instanceof InsertIntoHadoopFsRelationCommand) {
-      InsertIntoHadoopFsRelationCommand insert = (InsertIntoHadoopFsRelationCommand)logical;
-      return ImmutableList.of(insert.outputPath().toUri().toString());
-    } else {
-      extraInfo.append("No Output found for plan: ").append(logical.getClass().getName()).append(" ").append(logical).append("\n");
-      return ImmutableList.of();
+  @Override
+  public void setActiveJob(ActiveJob activeJob) {
+  }
+
+  @Override
+  public void start(SparkListenerJobStart jobStart) {
+    log.info("Starting job as part of spark-sql:" + jobStart.jobId());
+    QueryExecution queryExecution = SQLExecution.getQueryExecution(executionId);
+    if (queryExecution == null) {
+      log.info("No execution info {}", queryExecution);
+      return;
     }
+    LineageEvent event = LineageEvent.builder()
+        .inputs(buildInputs(queryExecution.logical()))
+        .outputs(buildOutputs(queryExecution.logical()))
+        .run(buildRun(
+            buildLogicalPlanFacet(queryExecution.logical()),
+            buildPhysicalPlanFacet(queryExecution.executedPlan())))
+        .job(buildJob())
+        .eventTime(ZonedDateTime.now())
```

Comment:
I think this is fine. The SparkListenerJobStart has a timestamp as well I believe. `startEvent.time()`